extensions/blocks/map/edit.js

```diff
@@ -185,6 +212,51 @@ class MapEdit extends Component {
 						] }
 					/>
 					<PanelBody title={ __( 'Map Settings', 'jetpack' ) }>
+						<BaseControl
+							label={ __( 'Height in pixels' ) }
+							id={ `block-jetpack-map-height-input-${ instanceId }` }
+						>
+							<input
+								type="number"
+								id={ `block-jetpack-map-height-input-${ instanceId }` }
+								onChange={ event => {
+									let height = parseInt( event.target.value, 10 );
+									if ( isNaN( height ) ) {
+										// Set map height to default size and input box to empty string
+										height = null;
+									} else if ( height < 400 ) {
+										// Set map height to minimum size
+										height = 400;
+									}
+									setAttributes( {
+										mapHeight: height,
+									} );
+									setTimeout( this.mapRef.current.sizeMap, 0 );
+								} }
+								value={ mapHeight || '' }
```

Comment:
I've noticed that the default map height is `420`.
The first time you click on either of this input's arrows when the value is empty, the value becomes `400` regardless, and the map size shrinks a little bit.

![Feb-17-2020 12-36-12](https://user-images.githubusercontent.com/2070010/74654406-206f8800-5182-11ea-9046-dde220d8f2cd.gif)

I find it a bit weird, especially when clicking on the increase arrow.

What about something like this?

```jsx
const DEFAULT_HEIGHT = 420;
const MIN_HEIGHT = 400;

<input
	value={ mapHeight || DEFAULT_HEIGHT }
	onChange={ event => {
		let height = parseInt( event.target.value, 10 );
		if ( isNaN( height ) || DEFAULT_HEIGHT === height ) {
			height = null;
		} else if ( height < MIN_HEIGHT ) {
			height = MIN_HEIGHT;
		}
		setAttributes( { mapHeight: height } );
		setTimeout( this.mapRef.current.sizeMap, 0 );
	} }
/>
```

Reply:
The default map height changes based on the window width: you can see this by resizing the editor window.

1dae92a52 uses the current map height, however it's calculated, as the value to insert when changing the height from default, using the incr/decr buttons. There are a couple of slightly weird things about this:

- Clicking the incr/decr buttons for the first time won't change the height of the map. I think this works to show the post author what they've done, so I'm not too concerned about it. 
- The first value it shows may not be divisible by 10, which means the next incr/decr click will just round it up/down. Slightly weird, but I don't think it's a deal breaker.

Reply:
Also worth noting: The `ResizableBox` handle snaps to a 10px grid: the first move of it will snap it to the grid, additional resizing will move in 10px increments/decrements.

Reply:
I came here to mention something similar; I got confused when I first clicked on the field and it went from empty to a default as soon as I tried typing something. I wanted to enter `600`, but it ended up entering `43500` because it switched to `435` when I typed `6`, and then carried on with the 0s I added.

When I saw that, I tried to hit backspace to remove everything and type 600 again, but I could not do that; I only allowed me to delete to 400. 

As it is, I don't know how to set the height to 600, short of using the little arrows.

Reply:
Ugh, I forgot that React triggers the `change` event when it receives an `input` event.

Fixed in 28cdc95eb.